nemo_curator/filters/synthetic.py

```diff
@@ -0,0 +1,219 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+from typing import List, Union
+
+import dask
+import dask.dataframe as dd
+import numpy as np
+import pandas as pd
+from dask.base import normalize_token, tokenize
+from openai import OpenAI
+
+from nemo_curator.filters.doc_filter import DocumentFilter
+from nemo_curator.utils.decorators import batched
+
+
+# ----------------------------------------------------------------------------80
+# ------------------------ EASINESS FILTER -------------------------------------
+# ----------------------------------------------------------------------------80
+class EasinessFilter(DocumentFilter):
+    """
+    Discards questions that are deemed easy to retrieve by retriever modls
+    """
+
+    def __init__(
+        self,
+        base_url: str = None,
```

Comment:
why are these parameters defaulted to `None`, aren't they required for the filter to work?